traffic_ops/traffic_ops_golang/api/api.go

```diff
@@ -369,6 +371,33 @@ func (inf *APIInfo) Close() {
 	}
 }
 
+// SendMail is a convenience method used to call SendMail using an APIInfo structure's configuration.
+func (inf *APIInfo) SendMail(to rfc.EmailAddress, msg []byte) (int, error, error) {
+	return SendMail(to, msg, inf.Config)
+}
+
+// SendMail sends an email msg to the address identified by to. The msg parameter should be an
+// RFC822-style email with headers first, a blank line, and then the message body. The lines of msg
+// should be CRLF terminated. The msg headers should usually include fields such as "From", "To",
+// "Subject", and "Cc". Sending "Bcc" messages is accomplished by including an email address in the
+// to parameter but not including it in the msg headers.
+// The cfg parameter is used to set things like the "From" field, as well as for connection
+// and authentication with an external SMTP server.
+// SendMail returns (in order) an HTTP status code, a user-friendly error, and an error fit for
+// logging to system error logs. If either the user or system error is non-nil, the operation failed,
+// and the HTTP status code indicates the type of failure.
+func SendMail(to rfc.EmailAddress, msg []byte, cfg *config.Config) (int, error, error) {
+	if !cfg.SMTP.Enabled {
+		return http.StatusServiceUnavailable, errors.New("SMTP is not enabled!"), nil
+	}
+	auth := smtp.PlainAuth("", cfg.SMTP.User, cfg.SMTP.Password, strings.Split(cfg.SMTP.Address, ":")[0])
```

Comment:
do you need to check that cfg.ConfigTO.EmailFrom is defined or not? Unsure if that causes an error in the SendMail func or not

Reply:
No, that field is mandatory in the configuration. If it doesn't exist, Traffic Ops won't start.

Reply:
can you point me to where it is marked as required and forces Traffic Ops not to start?

Reply:
I think it's probably most clearly illustrated on [Line #175 in `traffic_ops/traffic_ops_golang/config/config.go`](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/config/config.go#L175). If the config fails to load startup is blocked. ConfigTO and its sub-fields are mandatory, so their absence would cause a failure to load.

Reply:
I dont think that makes sure that the fields are defined it just does an unmarshal.

If we want to make it required to be defined doesn't it need to go in https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/config/config.go#L249

Reply:
Yes, you're right. I thought if the JSON parser didn't find it and the passed pointer wasn't a pointer to a pointer (so it can't set the underlying pointer to `nil`) then it would throw an error. But that's not true at all.

Reply:
So I fixed this by making ConfigTO and its children required parts of the config, which ought to be fine because Perl always required it anyway.